pages/p/[id].tsx

```diff
@@ -5,7 +5,7 @@ import Layout from "../../components/Layout"
 import { PostProps } from "../../components/Post"
 
 export const getServerSideProps: GetServerSideProps = async ({ params }) => {
-  const post = await prisma.post.findUnique({
+  const post = await prisma.post.findUniqueOrThrow({
```

Comment:
It would be helpful to test for the behaviour of the page when this function throws.

Reply:
From what the docs are saying this function does not throw an error at all, it's just going to return an object even if it's an empty object. link: https://www.prisma.io/docs/orm/reference/prisma-client-reference#finduniqueorthrow

Reply:
Those docs also say `However, if the query does not find a record, it returns NotFoundError: No User found error.` Can we test for this?

Reply:
I misread that paragraph. Yes, we can test it. when running this with no records, it renders the static 500 error page that next has. Underneath is the test that I wrote. This doesn't render the 500 error page how should I approach this to get my result?

```
it('Throws error when props is not found', async () => {
    global.prisma.post.findUniqueOrThrow.mockImplementation(() => {
      throw new Error('NotFoundError: No User found error')
    })
    const result = getServerSideProps(params)
    mockRouter.push('/pages/p/1')
    render(<Page {...result.props} />)
    await expect(result).rejects.toThrow('NotFoundError: No User found error')
  })
```

Reply:
First, check what page is rendered to debug who the exception is handled. Then, I think the ideal behaviour would be to catch the 500 errors and return a Post not found page (404). Check out this [link] for a guide.